pisek/task_jobs/data/testcase_info.py

```diff
@@ -48,11 +48,31 @@ def mixed(name: str) -> "TestcaseInfo":
     def static(name: str) -> "TestcaseInfo":
         return TestcaseInfo(name, 1, TestcaseGenerationMode.static, False)
 
-    def input_path(self, env: Env, seed: Optional[int] = None) -> TaskPath:
+    def input_path(
+        self, env: Env, seed: Optional[int] = None, solution: Optional[str] = None
+    ) -> InputPath:
         filename = self.name
         if self.seeded:
             assert seed is not None
             filename += f"_{seed:x}"
         filename += ".in"
 
-        return TaskPath.input_path(env, filename)
+        return InputPath(env, filename, solution=solution)
+
+    def reference_output(
+        self, env: Env, seed: Optional[int] = None, solution: Optional[str] = None
+    ) -> OutputPath:
+        is_static = self.generation_mode == TestcaseGenerationMode.static
+
+        input_path = self.input_path(env, seed, solution=env.config.primary_solution)
+        if is_static:
+            path = OutputPath.static(input_path.replace_suffix(".out").name)
+        else:
+            path = input_path.to_output()
```

Comment:
What's different about static outputs?

Reply:
Refernce static outputs are static, where others are generated by primary solution.

Reply:
Does this function return the path for a solution-specific copy of the reference output, or to a single shared instance of the reference output?

Reply:
Specific solution copy if `solution` is not `None`, otherwise `_inputs` instance.

Reply:
I personaly found this functions logic to be hard to follow, but I think I understand it now.